src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -116,79 +118,92 @@ func logHostsStatus(log logrus.FieldLogger, hosts map[string]inventory_client.Ho
 	log.Infof("Hosts status: %v", hostsStatus)
 }
 
-func (c *controller) WaitAndUpdateNodesStatus(status *ControllerStatus) {
+func (c *controller) WaitAndUpdateNodesStatus(ctx context.Context, wg *sync.WaitGroup) {
+	approveCtx, approveCancel := context.WithCancel(ctx)
+	defer func() {
+		approveCancel()
+		c.log.Infof("WaitAndUpdateNodesStatus finished")
+		wg.Done()
+	}()
+	// starting approve csrs
+	go c.ApproveCsrs(approveCtx)
 
 	c.log.Infof("Waiting till all nodes will join and update status to assisted installer")
+	_ = utils.WaitForPredicateWithContext(ctx, LongWaitTimeout, GeneralWaitInterval, c.waitAndUpdateNodesStatus)
```

Comment:
Why did you increase the wait for 3 hours? This logic runs prior to operators updates and the condition should be met in reasonable time. You will get a wait-for-control-plane timeout much sooner from the service which will result in terminating the thread all together

Reply:
till now there was no timeout for this one. 
I decreased timeout from run_for_ever to 10 hours